src/test/java/com/spotify/futures/FuturesExtraTest.java

```diff
@@ -571,4 +572,78 @@ public void testAsyncException6() throws Exception {
             });
     future.get();
   }
+
+  @Test
+  public void testWaitForNoInputs() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future);
+    assertSame(future, result);
+    assertTrue(result.isDone());
+
+  }
+
+  @Test
+  public void testWaitForOneInput() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1);
+
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test
+  public void testWaitForTwoInputs() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+    final ListenableFuture<String> other2 = Futures.immediateFuture("other2");
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1, other2);
+
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test
+  public void testWaitForTwoInputsOneFailure() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+    final ListenableFuture<String> other2 = Futures.immediateFailedFuture(new RuntimeException());
+    final ListenableFuture<String> other3 = Futures.immediateFuture("other2");
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1, other2, other3);
+
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test
+  public void testWaitForNotImmediate() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+    final ListenableFuture<String> other2 = Futures.immediateFailedFuture(new RuntimeException());
+    final SettableFuture<String> other3 = SettableFuture.create();
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1, other2, other3);
+
+    assertFalse(result.isDone());
+    other3.set("foo");
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test(expected = ExecutionException.class)
+  public void testWaitForNotImmediateWithFailure() throws Exception {
```

Comment:
For that usecase I would simply use FuturesExtra.syncTransformN or a transform with Futures.allAsList (which will fail if any waited fails)

This is only useful if you want to get some sort of ordering guarantees but don't want to be affected by other futures.